### PR TITLE
fix(infra-monitoring): state + url desync on quick filters

### PR DESCRIPTION
**Pull request by H4ad**
## Pull Request

---

Quick Filters, URL state and Current Query were not in sync.

After:

> URL State being in sync

https://github.com/user-attachments/assets/c7a05f7e-6ff8-42ba-8b34-3fb95d119f83

> Show All adding the filter to the query search & updating payload

https://github.com/user-attachments/assets/73029738-b9e1-4f44-b090-eada447befce

Related https://github.com/SigNoz/engineering-pod/issues/3997

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [x] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

https://github.com/SigNoz/engineering-pod/issues/3997

The issue was caused by missing sync between current query and URL state.

Add code to do the sync between URL and current query.

---

- Tests added/updated: -
- Manual verification: Yes
- Edge cases covered: -

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Bug Fix |
| Description | Query Filters on Infra Monitoring now are reflecting the real state on URL |

---

### 📋 Checklist
- [ ] Tests added or explicitly not required
- [x] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

### CODE REVIEW

File: frontend/src/container/InfraMonitoringK8s/Base/K8sBaseList.tsx [hunk 1]
```diff
@@ -28,9 +28,9 @@ import { openInNewTab } from 'utils/navigation';
 import { InfraMonitoringEntity } from '../constants';
 import {
 	useInfraMonitoringCurrentPage,
+	useInfraMonitoringFilters,
 	useInfraMonitoringGroupBy,
 	useInfraMonitoringOrderBy,
-	useInfraMonitoringQueryFilters,
```
Comment (AshwinBhatkal) [inline frontend/src/container/InfraMonitoringK8s/Base/K8sBaseList.tsx:33]: is this deprecated or is this different?

File: frontend/src/container/InfraMonitoringK8s/Base/K8sHeader.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ import {
 } from '../constants';
 import {
 	useInfraMonitoringCurrentPage,
-	useInfraMonitoringFiltersK8s,
```
Comment (AshwinBhatkal) [inline frontend/src/container/InfraMonitoringK8s/Base/K8sHeader.tsx:23]: same Q as previous here

**Review by AshwinBhatkal — APPROVED** [review]

File: frontend/src/container/InfraMonitoringK8s/Base/K8sBaseList.tsx (hunk 1 above)
Comment (H4ad) [reply frontend/src/container/InfraMonitoringK8s/Base/K8sBaseList.tsx:33]: It was the same thing with different name, removed all duplication and keep as one.

File: frontend/src/container/InfraMonitoringK8s/Base/K8sHeader.tsx (hunk 2 above)
Comment (H4ad) [reply frontend/src/container/InfraMonitoringK8s/Base/K8sHeader.tsx:23]: After merging hosts and k8s, we had 2 hooks for the filters, this just cleans and merge them.

File: frontend/src/container/InfraMonitoringK8s/Base/K8sBaseList.tsx (hunk 1 above)
Comment (H4ad) [reply frontend/src/container/InfraMonitoringK8s/Base/K8sBaseList.tsx:33]: And since now the source of truth is url, we can stop using current query, since it will be merged with current query eventually (this was the bug, it was not in the past)
